spring-boot-project/spring-boot/src/main/java/org/springframework/boot/web/reactive/context/ReactiveWebServerApplicationContext.java

```diff
@@ -66,9 +66,15 @@ public final void refresh() throws BeansException, IllegalStateException {
 			super.refresh();
 		}
 		catch (RuntimeException ex) {
-			WebServerManager serverManager = this.serverManager;
-			if (serverManager != null) {
-				serverManager.getWebServer().stop();
+			try {
+				WebServer webServer = getWebServer();
+				if (webServer != null) {
+					webServer.stop();
+					webServer.destroy();
+				}
+			}
+			catch (Exception shutdownEx) {
+				ex.addSuppressed(shutdownEx);
```

Comment:
Let's consider this separately please. If we do it, it should be applied on the servlet side as well.

Reply:
Sure, I've reverted these changes.

Reply:
We think that adding the stop or destroy failure as a suppressed exception is a good idea. Would you like to open a PR for it or would you prefer that we take care of it?

Reply:
[#44310](https://github.com/spring-projects/spring-boot/pull/44310)

Reply:
Thanks!